[CQT-189] [Formatting] Add clang-format (#256)
QuTech-Delft/libqasm: docs/dev-guide/dev-guide.md

    - `mkdocstrings`
    - `pymdown-extensions`

### Linters

- `clang-format-18`
- `clang-tidy-18`

> In a Linux machine, these linters can be installed with the following commands:

Reviewer: ```suggestion
On a Linux machine, these linters can be installed with the following commands:
```

### Options

> Profiles are a shorthand for command line options. The command above could be written, similarly, as:

Reviewer: ```suggestion
Profiles are a shorthand for command line options.
The command above could be written, similarly, as:
```

Author: These two sentences fit perfectly on a line. So discarding this suggestion.

Reviewer: Hmmm, this goes against what you wanted for cQASM spec right? You wanted to at least have each separate sentence on a new line. Or did I misunderstand...

    The `export` is needed to point `mkdocs` to the custom handlers used for the C++, emscripten, and Python APIs.

## Linters

> Continuous Integration will fail if the files don't adhere to a series of formatting and code style guidelines:

Reviewer: ```suggestion
Continuous Integration will fail if the files do not adhere to a series of formatting and code style guidelines:
```

- Formatting checks are defined in `.clang-format`.
- Code style checks are defined in `.clang-tidy`.

It is recommended to run these linters before pushing any change:

```shell
python3 ./scripts/run_cpp_linters.py .
```

!!! note

>     The linters require `clang-format-18` and `clang-tidy-18` to be installed in the system.

Reviewer: ```suggestion
    The linters require `clang-format-18` and `clang-tidy-18` to be installed on the system.
```
